### PR TITLE
Improving starting utask on dev

### PR DESCRIPTION
**Pull request by rbeuque74**
This pull-request introduces script wait-for-it.sh to avoid utask binary starts before database is ready.

Previous behavior: 

```
romain@laptop:~/utask‼> docker-compose up
Creating network "utask_default" with the default driver
Creating utask_db_1 ... done
Creating utask_utask_1 ... done
Attaching to utask_db_1, utask_utask_1
utask_1  | time="2019-11-16T16:38:04Z" level=info msg="[DatabaseConfig] Using 50 max open connections, 30 max idle connections, 60 seconds timeout"
utask_1  | time="2019-11-16T16:38:04Z" level=fatal msg="dial tcp 172.22.0.2:5432: connect: connection refused"
db_1     | The files belonging to this database system will be owned by user "postgres".
db_1     | This user must also own the server process.
db_1     | 
db_1     | The database cluster will be initialized with locale "en_US.utf8".
db_1     | The default database encoding has accordingly been set to "UTF8".
db_1     | The default text search configuration will be set to "english".
db_1     | 
db_1     | Data page checksums are disabled.
db_1     | 
db_1     | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1     | creating subdirectories ... ok
db_1     | selecting default max_connections ... 100
db_1     | selecting default shared_buffers ... 128MB
db_1     | selecting dynamic shared memory implementation ... posix
db_1     | creating configuration files ... ok
db_1     | creating template1 database in /var/lib/postgresql/data/base/1 ... ok
db_1     | initializing pg_authid ... ok
db_1     | initializing dependencies ... ok
db_1     | creating system views ... ok
db_1     | loading system objects' descriptions ... ok
db_1     | creating collations ... ok
db_1     | creating conversions ... ok
db_1     | creating dictionaries ... ok
db_1     | setting privileges on built-in objects ... ok
db_1     | creating information schema ... ok
db_1     | loading PL/pgSQL server-side language ... ok
utask_utask_1 exited with code 1
```

New behavior:
```
romain@maestro:~/utask$ docker-compose up
Creating network "utask_default" with the default driver
Creating utask_db_1 ... done
Creating utask_utask_1 ... done
Attaching to utask_db_1, utask_utask_1
db_1     | The files belonging to this database system will be owned by user "postgres".
....
utask_1  | wait-for-it.sh: waiting 15 seconds for db:5432
db_1     | creating subdirectories ... ok
...
db_1     | LOG:  incomplete startup packet
utask_1  | wait-for-it.sh: db:5432 is available after 3 seconds
utask_1  | time="2019-11-16T16:46:19Z" level=info msg="[DatabaseConfig] Using 50 max open connections, 30 max idle connections, 60 seconds timeout"
utask_1  | time="2019-11-16T16:46:19Z" level=info msg="Created task template 'hello-world-now'"
utask_1  | [GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.
```

Also switch base image from golang:1.12.7 to golang:1.12 (pointing on latest build from 1.12, currently 1.12.13) to embed last Golang security fixes.

Also saving 2 docker layers :)

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ services:
       - 8081:8081
     volumes:
       - "./templates:/app/templates:ro"
+    depends_on:
+      - db
+    command: ["/wait-for-it.sh", "db:5432", "--", "/app/utask"]
```
Comment (w3st3ry) [inline docker-compose.yaml]: Could you just move the `command:` key just after the `build:` key (L.4) to be more consistent with the common way of writing a `docker-compose`?

File: Dockerfile [hunk 2]
```diff
@@ -11,21 +11,21 @@ RUN BASEHREF=/ui/dashboard/ PREFIX_API_BASE_URL=/ make build-prod
...
-
-RUN mkdir -p /app/plugins /app/templates /app/config /app/init /app/static/dashboard /app/static/editor
+RUN make re && \
+    mv hack/Makefile-child Makefile && \
+    mkdir -p /app/plugins /app/templates /app/config /app/init /app/static/dashboard /app/static/editor
 WORKDIR /app
 
 COPY --from=js-builder /home/node/ui/dashboard/dist/utask-ui/*  /app/static/dashboard/
 COPY --from=js-builder /home/node/ui/editor/dist/utask-editor/* /app/static/editor/
+ADD https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh /wait-for-it.sh
```
Comment (Alkorin) [inline Dockerfile]: Would be better to vendor this script.
Who knows what will be the content of this shell script later ;)

File: docker-compose.yaml (hunk 1 above)
Comment (rbeuque74) [reply docker-compose.yaml]: Done, thanks

File: Dockerfile (hunk 2 above)
Comment (rbeuque74) [reply Dockerfile]: Done, thanks!

File: Dockerfile [hunk 3]
```diff
@@ -11,21 +11,21 @@ RUN BASEHREF=/ui/dashboard/ PREFIX_API_BASE_URL=/ make build-prod
...
 COPY .  /go/src/github.com/ovh/utask
 WORKDIR /go/src/github.com/ovh/utask
-RUN make re
-RUN mv hack/Makefile-child Makefile
-
-RUN mkdir -p /app/plugins /app/templates /app/config /app/init /app/static/dashboard /app/static/editor
+RUN make re && \
+    mv hack/Makefile-child Makefile && \
+    mkdir -p /app/plugins /app/templates /app/config /app/init /app/static/dashboard /app/static/editor && \
+    mv hack/wait-for-it/wait-for-it.sh /wait-for-it.sh
```
Comment (Alkorin) [inline Dockerfile]:
```suggestion
    mv hack/wait-for-it/wait-for-it.sh /wait-for-it.sh && chmod +x /wait-for-it.sh
```

File: Dockerfile [hunk 4]
```diff
@@ -11,21 +11,21 @@ RUN BASEHREF=/ui/dashboard/ PREFIX_API_BASE_URL=/ make build-prod
...
 WORKDIR /app
 
 COPY --from=js-builder /home/node/ui/dashboard/dist/utask-ui/*  /app/static/dashboard/
 COPY --from=js-builder /home/node/ui/editor/dist/utask-editor/* /app/static/editor/
 
-RUN cp /go/src/github.com/ovh/utask/utask /app/utask
-RUN chmod +x /app/utask
+RUN cp /go/src/github.com/ovh/utask/utask /app/utask && \
+    chmod +x /app/utask /wait-for-it.sh
```
Comment (Alkorin) [inline Dockerfile]:
```suggestion
    chmod +x /app/utask
```

File: Dockerfile (hunk 3 above)
Comment (Alkorin) [reply Dockerfile]: To leave together the commands associated with wait-for-it

**Pull request merged**